agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannel.java

```diff
@@ -389,6 +413,21 @@ private Consumer<Throwable> errorHandler(
     };
   }
 
+  private void statsbeatDidNotReachBreeze() {
+    if (statsbeatUnableToReachBreezeCounter.getAndIncrement() >= 10
+        && !statsbeatHasBeenShutdown.getAndSet(true)) {
```

Comment:
we never set statsbeatHasBeenShutdown to false.  Like @heyams mentioned above, this boolean seems useless.

Reply:
this line that you have highlighted sets the boolean, it is not useless

Reply:
@heyams comment was that it was useless if we implement her suggestion above of shutting down statsbeat retries from disk (which I'd like to defer until `vnext` branch to limit merge conflicts at this point)

Reply:
Is [line](https://github.com/microsoft/ApplicationInsights-Java/blob/0dfa662f055e3dc3a54e74f5c1cc3fb9a5eb5ab3/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannel.java#L127) the place where we retry from disk?  Sorry, it wasn't obvious for me that that is what sendRawBytes did.  If that is the case then this is all good.  Ship it!!